### ISSUE TITLE
Disconnect 127.0.0.1:6385 because the node does not hold any slot

### ISSUE THREAD

**Issue by igorlimansky** [luin/ioredis#935]
I'm experiencing an issue while trying to connect to redis-cluster which is port-forwarded to localhost from kubernetes ( redis cluster deployed via kubedb redis 3 masters each have 1 slave )
https://pastebin.com/5vy64G8J

here is how I'm trying to connect to it from code:
```
client = new Redis.Cluster(
  nodes.map(n => {
    const [ip, port] = n.split(':')
    return ({
      host: ip,
      port,
    })
  }),
  {
    enableOfflineQueue: true,
    natMapping: {
      '10.12.5.52:6379': { host: 'localhost', port: 6380 },
      '10.12.10.28:6379': { host: 'localhost', port: 6381 },
      '10.12.3.29:6379': { host: 'localhost', port: 6382 },
      '10.12.5.53:6379': { host: 'localhost', port: 6383 },
      '10.12.5.54:6379': { host: 'localhost', port: 6384 },
      '10.12.5.55:6379': { host: 'localhost', port: 6385 },
    },
    dnsLookup: async (address, callback) => {
      // const ip = await dns.lookupAsync(address)
      // console.log(address, '=>', ip)
      callback(null, address)
    },
  })
```

even if I remove NAT mapping - logs/errors are still the same
**upd:** increasing`slotsRefreshTimeout` doesnt help also,
 changing `localhost` to `127.0.0.1` didnt help too

```
/data # redis-cli -c cluster slots
1) 1) (integer) 0
   2) (integer) 5460
   3) 1) "10.12.10.56"
      2) (integer) 6379
      3) "89689b18c44b74a75d0084ae87e8cfd2e968d1a1"
   4) 1) "10.12.5.88"
      2) (integer) 6379
      3) "170835484932d492ed4cec66e4d1c29975ddb2ae"
2) 1) (integer) 10923
   2) (integer) 16383
   3) 1) "10.12.12.33"
      2) (integer) 6379
      3) "9f73dc3f93225d302e70ad5e662be6fd8c3222e6"
   4) 1) "10.12.3.49"
      2) (integer) 6379
      3) "d5cd41c28a605b93e22cf85759cd70708ac8b883"
3) 1) (integer) 5461
   2) (integer) 10922
   3) 1) "10.12.2.34"
      2) (integer) 6379
      3) "49fa10062ecb102071208d811f791c6513faeda5"
   4) 1) "10.12.11.43"
      2) (integer) 6379
      3) "9d04fdee6848fe66980c257c05ec7d7b01e931a3"
/data # exit
➜  kubectl exec -it redis-cluster2-shard0-0 -c redis -- sh
/data # redis-cli -c cluster slots
1) 1) (integer) 0
   2) (integer) 5460
   3) 1) "10.12.10.56"
      2) (integer) 6379
      3) "89689b18c44b74a75d0084ae87e8cfd2e968d1a1"
   4) 1) "10.12.5.88"
      2) (integer) 6379
      3) "170835484932d492ed4cec66e4d1c29975ddb2ae"
2) 1) (integer) 10923
   2) (integer) 16383
   3) 1) "10.12.12.33"
      2) (integer) 6379
      3) "9f73dc3f93225d302e70ad5e662be6fd8c3222e6"
   4) 1) "10.12.3.49"
      2) (integer) 6379
      3) "d5cd41c28a605b93e22cf85759cd70708ac8b883"
3) 1) (integer) 5461
   2) (integer) 10922
   3) 1) "10.12.2.34"
      2) (integer) 6379
      3) "49fa10062ecb102071208d811f791c6513faeda5"
   4) 1) "10.12.11.43"
      2) (integer) 6379
      3) "9d04fdee6848fe66980c257c05ec7d7b01e931a3"
/data # exit
➜  kubectl exec -it redis-cluster2-shard1-0 -c redis -- sh
/data # redis-cli -c cluster slots
1) 1) (integer) 5461
   2) (integer) 10922
   3) 1) "10.12.2.34"
      2) (integer) 6379
      3) "49fa10062ecb102071208d811f791c6513faeda5"
   4) 1) "10.12.11.43"
      2) (integer) 6379
      3) "9d04fdee6848fe66980c257c05ec7d7b01e931a3"
2) 1) (integer) 10923
   2) (integer) 16383
   3) 1) "10.12.12.33"
      2) (integer) 6379
      3) "9f73dc3f93225d302e70ad5e662be6fd8c3222e6"
   4) 1) "10.12.3.49"
      2) (integer) 6379
      3) "d5cd41c28a605b93e22cf85759cd70708ac8b883"
3) 1) (integer) 0
   2) (integer) 5460
   3) 1) "10.12.10.56"
      2) (integer) 6379
      3) "89689b18c44b74a75d0084ae87e8cfd2e968d1a1"
   4) 1) "10.12.5.88"
      2) (integer) 6379
      3) "170835484932d492ed4cec66e4d1c29975ddb2ae"
/data # redis-cli -c cluster info
cluster_state:ok
cluster_slots_assigned:16384
cluster_slots_ok:16384
cluster_slots_pfail:0
cluster_slots_fail:0
cluster_known_nodes:6
cluster_size:3
cluster_current_epoch:3
cluster_my_epoch:2
cluster_stats_messages_ping_sent:4520
cluster_stats_messages_pong_sent:4523
cluster_stats_messages_meet_sent:2
cluster_stats_messages_sent:9045
cluster_stats_messages_ping_received:4520
cluster_stats_messages_pong_received:4522
cluster_stats_messages_meet_received:3
cluster_stats_messages_received:9045
```

So basically error is that I got connected to cluster and then disconnect happens: Disconnect 127.0.0.1:6385 because the node does not hold any slot
And it tries to reconnect after timeout

**Comment by igorlimansky** [comment]
@luin hello, I'm still missing connection to redis cluster and soon it will become a problem for me, do you have any idea why such error can happen?

**Comment by ChechaTocon** [comment]
Hello @devops398, did you find the solution for this problem?, I have the same issue, but I found out that if I connect with TLS enabled it happens, I can connect without TLS. 

Once I add the 
      ```
    redisOptions: {
            tls:{},   <--TLS enabled
            password: cachePassword,
          },
```

It says:
[ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames

So I add the dnsLookup:

`var redis = new Redis.Cluster(
        [
          {
            host: cacheHostName,
            port: 6380,            
          },
        ],
        {
          //slotsRefreshTimeout: 2000,
          //scaleReads: "slave",
          enableOfflineQueue: true,
          dnsLookup: (address, callback) =>
            callback(null, address).then(),
          redisOptions: {
            tls:{},
            password: cachePassword,
          },
          retryStrategy: function retryStrategy(times) {
            if (times == 3) {
              // redis.connect()
              return new Error("Redis Retry time exhausted");
            }
            const delay = Math.min(times * 50, 2000);
            return delay;
          },
        }
      );`

But now the error is the following:

Disconnect <hostname>:6380 because the node does not hold any slot

**Comment by 0xrinegade** [comment]
hey @ChechaTocon no i didnt
i tried with/without tls everything tbh, i got lucky that even one redis instance managed to process all binance trades in realltime with decent orderbook depth so i left it unscalable
